addons/product/models/product_template.py

```diff
@@ -964,6 +964,18 @@ def _create_product_variant(self, combination, log_warning=False):
             'product_template_attribute_value_ids': [(6, 0, combination._without_no_variant_attributes().ids)]
         })
 
+    def _create_first_product_variant(self, log_warning=False):
+        """Create if necessary and possible and return the first product
+        variant for this template.
+
+        :param log_warning: whether a warning should be logged on fail
+        :type log_warning: bool
+
+        :return: the first product variant or none
+        :rtype: recordset of `product.product`
+        """
+        return self._create_product_variant(self._get_first_possible_combination(), log_warning)
```

Comment:
This is new diff :eyes:
Either you move the method and use it or you don't move it, but don't move it for no specific reason ;)

Reply:
This is what seb suggested. So `_create_first_product_variant` should be defined in the parent module `product` instead of `website_sale`, should I make a PR for it? I add it to this one cause I found it's also appears in V13

Reply:
Indeed, it should be defined in product, but why don't you use it in `copy` ?
`return template.product_variant_id or template._create_first_product_variant()`

Reply:
Great point!